src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -82,15 +83,24 @@ class LogInWithShortLivedTokenPage extends Component {
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
         const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
 
-        // User is signing in to a different account sign them out and sign in again
-        if (email !== this.props.session.email) {
+        const signedIn = this.props.session && this.props.session.authToken;
+        if (!signedIn) {
             const accountID = parseInt(lodashGet(this.props.route.params, 'accountID', ''), 10);
             const shortLivedToken = lodashGet(this.props.route.params, 'shortLivedToken', '');
-            Session.signOut();
             Session.signInWithShortLivedToken(accountID, email, shortLivedToken);
             return;
         }
 
+        // User is trying to transition with a different account than the one
+        // they are currently signed in as so we will sign them out, clear Onyx,
+        // and cancel all network requests. This component will mount again from
+        // PublicScreens and since they are no longer signed in, a request will be
+        // made to sign them in with their new account.
+        if (email !== this.props.session.email) {
+            Session.signOutAndRedirectToSignIn();
+            return;
+        }
```

Comment:
Small idea on how to improve this. I noticed that the comment here is pretty similar to the one in `componentDidMount()`. What if we did something like this...

```js
if (this.signOutIfNeeded(email)) {
    return;
}
```

And then wrote a method that does this

```js
// add the long comment here in the JSDoc...
signOutIfNeeded(email) {
    if (email === this.props.session.email) {
        return false;
    }

    Session.signOutAndRedirectToSignIn();
    return true;
}
```

Reply:
That's a great idea, I'll do that.

Reply:
I also extracted the sign in logic to a function because it was also repeated twice.